spec/storage_spec.rb

```diff
@@ -16,6 +16,21 @@
   let(:storage) { Faraday::HttpCache::Storage.new(store: cache) }
   subject { storage }
 
+  before do
+    allow(Kernel).to receive(:warn).with(
+      'Deprecated: Faraday::HttpCache::Storage is deprecated and will be removed '\
+      'in the next major release. Use Faraday::HttpCache::Strategies::ByUrl instead.'
+    )
+  end
+
+  it 'creates strategy and warns about deprecation' do
+    expect(Kernel).to receive(:warn).with(
+      'Deprecated: Faraday::HttpCache::Storage is deprecated and will be removed '\
+      'in the next major release. Use Faraday::HttpCache::Strategies::ByUrl instead.'
+    )
+    expect(storage).to be_a_kind_of(Faraday::HttpCache::Strategies::ByUrl)
```

Comment:
**nitpick (non-blocking)**: You can use [**One-liner** syntax ](https://relishapp.com/rspec/rspec-core/docs/subject/one-liner-syntax) expectation because `storage` is the `subject`.
```suggestion
    is_expected.to be_a_kind_of(Faraday::HttpCache::Strategies::ByUrl)
```